homeassistant/components/steam_online/config_flow.py

```diff
@@ -0,0 +1,222 @@
+"""Config flow for Steam integration."""
+from __future__ import annotations
+
+import functools as ft
+from typing import Any
+
+import steam
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, Platform
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv, entity_registry as er
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_ACCOUNT, CONF_ACCOUNTS, DEFAULT_NAME, DOMAIN, LOGGER
+
+
+async def validate_input(user_input: dict, hass: HomeAssistant) -> list[dict[str, str]]:
+    """Handle common flow input validation."""
+    steam.api.key.set(user_input[CONF_API_KEY])
+    interface = await hass.async_add_executor_job(steam.api.interface, "ISteamUser")
+    names = await hass.async_add_executor_job(
+        ft.partial(interface.GetPlayerSummaries, steamids=user_input[CONF_ACCOUNT])
+    )
+    return (await hass.async_add_executor_job(names.get, "response"))["players"][
+        "player"
+    ]
+
+
+class SteamFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for team."""
+
+    def __init__(self) -> None:
+        """Initialize the flow."""
+        self.entry: config_entries.ConfigEntry | None = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> config_entries.OptionsFlow:
+        """Get the options flow for this handler."""
+        return SteamOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+        if user_input is None and self.entry:
+            user_input = {CONF_ACCOUNT: self.entry.data[CONF_ACCOUNT]}
+        elif user_input is not None:
```

Comment:
This will not trigger if user input was just set based on `self.entry` .

Reply:
That's correct. It's for reauth